### ISSUE TITLE
:lady_beetle: get() method in Decoder::decompress s/b in->get

### ISSUE THREAD

**Issue by pete4abw** [pete4abw/lrzip-next#117]
### lrzip-next Version

ZPAQ 7.15 library

### lrzip-next command line

N/A

### What happened?

For the longest time, ZPAQ could not decompress using zpaq levels 1 and 2. Only 3,4, and 5 levels were set in util.c. ZPAQ Decoder would report an error on decompression
```
Decompressing...
ZPAQ Error: missing end of segment marker:
```
Even though the archive was compressed properly.

### What was expected behavior?

ZPAQ compression should be able to utilize all 5 levels.

### Steps to reproduce

Modify util.c setup_overhead() function to use all 5 zpaq levels. Recompile, then encode and try and decode using level 1.
```
diff --git a/src/util.c b/src/util.c
index b8ad6d3..bb84c98 100644
--- a/src/util.c
+++ b/src/util.c
@@ -141,8 +141,8 @@ void setup_overhead(rzip_control *control)
                 * where default is 16KB */
                control->overhead = ((i64)control->dictSize * 23 / 2) + (6 * ONE_MB) + 16384;
        } else if (ZPAQ_COMPRESS) {
-               control->zpaq_level = (control->compression_level < 4 ? 3 :
-                                       (control->compression_level < 8 ? 4 : 5));
+               control->zpaq_level = control->compression_level * 5 / 9;
+               if (!control->zpaq_level) control->zpaq_level = 1;
                if (control->zpaq_bs == 0) {
                        switch (control->compression_level) {
                        case 1:
```

The fix will be in libzpaq.cpp. A simple omission to utilize the custom Reader class used by lrzip-next.
```
diff --git a/src/lib/libzpaq/libzpaq.cpp b/src/lib/libzpaq/libzpaq.cpp
index 7bf3673..d343f93 100644
--- a/src/lib/libzpaq/libzpaq.cpp
+++ b/src/lib/libzpaq/libzpaq.cpp
@@ -2148,7 +2148,7 @@ int Decoder::decompress() {
   }
   else {
     if (curr==0) {
-      for (int i=0; i<4; ++i) curr=curr<<8|get();
+      for (int i=0; i<4; ++i) curr=curr<<8|in->get();
       if (curr==0) return -1;
     }
     --curr;
```

### Please provide system details

OS Distro: Slackware
Kernel Version (uname -a): 6.3
System ram (free -h): lots

### Additional Context

After the changes...

Decompressing...
100%    9870.00 /   9870.00 KB  ZPAQ    1:100%  2:100%  
Average DeCompression Speed:  9.000MB/s
SHA256:79e741a75ce4c259e0d72048a064829d34c691335a5adbf4588330c3b52e9307
[OK] - 10,106,880 bytes                                

Total time: 00:00:00.08

**Comment by pete4abw** [comment]
Fixed in whats-next branch. Will migrate to main soon.

**Comment by pete4abw** [comment]
Fixed in lzma-22.01 v0.11.1
